consensus/src/density.rs

```diff
@@ -0,0 +1,221 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module implements window density APIs to aid in long range fork
+//! selection rule during chain selection in the mina consensus algorithm.
+
+use crate::error::ConsensusError;
+use mina_rs_base::consensus_state::ConsensusState;
+use mina_rs_base::numbers::BlockTime;
+use mina_rs_base::numbers::Length;
+
+// TODO: derive from protocol constants
+pub struct ConsensusConstants {
+    /// Point of finality (number of confirmations)
```

Comment:
So we can use the `ConsensusConstants` here without re-implement in the consensus crate?

Reply:
This is a good example of why refactoring to our own internal types that are computable from the WireTypes will be useful. This is effectively what is being done here anyway!

Reply:
Yeah, we really need to separate the wire types